src/main.rs

```diff
@@ -71,24 +74,68 @@ fn SupervisorTimer() {
 
 #[no_mangle]
 fn MachineExternal() {
-    sprint!(".");
+    let mut dr = unsafe {
+        DeviceResources::steal()
+    };
```

Comment:
Not for now, but I wonder what a safe architecture looks like for interrupt handling: I think we're giving up rust's data-race-freedom here.

Reply:
Yeah this is unsafe for a reason haha.

I wonder if we can do unsafe setup more up front, and then switch to a "safe" way to interact with the cpu? The general strategy that the hifive1 crate uses is to make structs that are essentially empty, with impls that then touch memory. The only way to get an instance is to go through the board instance, since there's a dummy private field of the struct.

We could do something like that, in conjunction with the phill opperman lazy_static! and spin lock? So we guaranteed only have one access at a time?

```
use spin::Mutex;
...
lazy_static! {
    pub static ref WRITER: Mutex<Writer> = Mutex::new(Writer {
        column_position: 0,
        color_code: ColorCode::new(Color::Yellow, Color::Black),
        buffer: unsafe { &mut *(0xb8000 as *mut Buffer) },
    });
}
```
from https://os.phil-opp.com/vga-text-mode/